### ISSUE TITLE
PHP 8 - `cache_expiration` option not being cast as int properly

### ISSUE THREAD

**Issue by cjg89** [UCF/UCF-RSS-Feed-Plugin#13]
**Describe the bug**

PHP 8.x starts throwing a fatal error in includes/ucf-rss-common.php on line 51:

PHP Fatal error:  Uncaught TypeError: Unsupported operand types: string * int in ucf-rss-feed-plugin/includes/ucf-rss-common.php:51

The line in question points to `UCF_RSS_Common::get_cache_expiration()`:
https://github.com/UCF/UCF-RSS-Feed-Plugin/blob/debd29bff2fc3a354ec51fea6fe48e154e324eaa/includes/ucf-rss-common.php#L43-L52

which suggests that the `cache_expiration` is not being properly cast as an int (see https://github.com/UCF/UCF-RSS-Feed-Plugin/blob/master/includes/ucf-rss-config.php#L153-L161)

**Steps to Reproduce**

n/a (reported on Teams)

**Expected Behavior**

`cache_expiration` should be an integer, and we should be able to multiply its value by `HOUR_IN_SECONDS`.

**Possible Fix**

n/a

**Your environment:**

WP 5.6, PHP 8, Plugin version 1.0.2

**Additional context**

Relevant discussion: https://teams.microsoft.com/l/message/19:5325780226474b76b53d50029eae9e22@thread.skype/1617983433732?tenantId=bb932f15-ef38-42ba-91fc-f3c59d5dd1f1&groupId=5e3e72e2-3599-47b3-bcfc-bb3d5b579f12&parentMessageId=1617983433732&teamName=UCFTeam-CAMPUS-Wordpress&channelName=help-plugins&createdTime=1617983433732

**Comment by bagofarms** [comment]
We need to update our sites to PHP 8, so I'm trying to address this issue now. It looks like the root of the problem is that [ucf-rss-config.php Line 483](https://github.com/UCF/UCF-RSS-Feed-Plugin/blob/master/includes/ucf-rss-config.php#L483)
```php
return get_option( $option_name, $option->get_default() );
```
returns an empty string. I believe this is because `ucf_rss_cache_expiration` is not found.

Additionally, there is no method that matches that signature, since `UCF_RSS_Option::get_option` does not take a second parameter. This same invalid signature is used in [get_default()](https://github.com/UCF/UCF-RSS-Feed-Plugin/blob/debd29bff2fc3a354ec51fea6fe48e154e324eaa/includes/ucf-rss-config.php#L43) as well. I'm not certain what the purpose of these calls is, but it appears that you are trying to get the `default` property of those objects.
I propose changing [ucf-rss-config.php Line 483](https://github.com/UCF/UCF-RSS-Feed-Plugin/blob/master/includes/ucf-rss-config.php#L483) to:
```php
return $option->default;
```
I've tested this on our QA server, and it appears to be working so far, but I'm not sure if I'm skipping some vital part of the code. I'll continue testing it on our end, and create a PR if I feel like it's working as expected. I would appreciate some feedback on my thought process, though.

**Comment by jmbarne3** [comment]
Sorry for the delay on this - I wasn't even aware this was an outstanding issue until you commented yesterday. This ended up being a fairly simple issue: The cache expiration setting didn't have a section assigned to it so it would show up in the options page, therefore the option wasn't actually configurable via the GUI. However, that doesn't mean it wasn't getting saved if someone saved the options for including CSS or the fallback image. And when it was saved, it was being saved as an empty string. So the function call is working correctly, in that if there is a value set in the database, it should format the value and return it instead of the default. In this case, however, the value being returned from the database was an empty string, which is meaningless in this context.

What I am going to do is (1) make sure the setting is editable on the options screen so it can be adjusted and (2) ensure the value returned by default is always some sort of integer, even if the database is storing an empty string.

**Comment by bagofarms** [comment]
Thanks for getting back to me so quickly! You explanation makes sense. I noticed it was going through the same process as other options that were configurable in the GUI, but I couldn't figure out why it was behaving the way it was. I'll keep using my workaround for now, and I will update to the fixed version as soon as it is available. Let me know if you need any assistance.

**Comment by jmbarne3** [comment]
I have the code ready, but I'm getting a memory error when running on php8 (which reset my default max memory back to 128mb) - Are you experiencing something similar? I'm wondering if this code has a memory leak somewhere.

**Comment by bagofarms** [comment]
I'm experiencing slow load times, but I haven't gotten a memory error with the default 120Mb. Our QA server isn't very fast, so I thought that was the reason.

**Comment by jmbarne3** [comment]
Closed by #27 and release version `v1.0.3`
